backend/src/main/java/com/skr/virtuallibrary/configurations/SecurityConfig.java

```diff
@@ -34,15 +34,18 @@ public SecurityFilterChain filterChain(HttpSecurity http,
                 .authorizeHttpRequests(auth ->
                         auth
                                 .requestMatchers("/swagger-ui/**", "/v3/api-docs/**").permitAll()
-                                .requestMatchers(HttpMethod.POST, "/auth/quick-register").hasAuthority(ADMIN_AUTHORITY)
-                                .requestMatchers("/auth/**").permitAll()
+                                .requestMatchers("/error").anonymous()
+
                                 .requestMatchers(HttpMethod.GET, "/books/**").permitAll()
+                                .requestMatchers(HttpMethod.GET, "/authors/**").permitAll()
+                                .requestMatchers(HttpMethod.GET, "/genres/**").permitAll()
                                 .requestMatchers(HttpMethod.GET, "/files/cover/**").permitAll()
-                                .requestMatchers(HttpMethod.POST, "/books/**").hasAuthority(ADMIN_AUTHORITY)
-                                .requestMatchers(HttpMethod.POST, "/authors/**").hasAuthority(ADMIN_AUTHORITY)
-                                .requestMatchers(HttpMethod.GET, "/reviews/**").permitAll()
-                                .requestMatchers("/error").anonymous()
-                                .anyRequest().authenticated()
+                                .requestMatchers("/reviews/**").permitAll()
```

Comment:
Walidacja użytkownika przeciwdziała temu, ale `anonymousUser` może wysłać np. POST request.
W sumie to tak też nie może być bo odpowiedź to: `404 NOT_FOUND "User could not be found with email: anonymousUser`, a raczej powinno być `401 Unauthorized`, więc i tak powinno się to wyciąć.